.github/workflows/on-pr.yaml

```diff
@@ -0,0 +1,54 @@
+name: Test CI/CD Pipeline
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+
+env:
+  CI: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: ["20.19.4"]
```

Comment:
Perhaps we could drop the matrix for now since we're testing with a single node version?